libs/traits/src/lib.rs

```diff
@@ -281,56 +285,103 @@ pub trait CurrencyPrice<CurrencyId> {
 	) -> Option<PriceValue<CurrencyId, Self::Rate, Self::Moment>>;
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum CompoundingSchedule {
+	/// Interest compounds every second
+	Secondly,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum InterestRate<Rate> {
+	/// Interest accrues at a fixed rate
+	Fixed {
+		rate_per_year: Rate,
+		compounding: CompoundingSchedule,
+	},
+}
+
+impl<Rate: EnsureAdd + EnsureSub> InterestRate<Rate> {
+	pub fn ensure_add(self, rate: Rate) -> Result<InterestRate<Rate>, ArithmeticError> {
+		match self {
+			InterestRate::Fixed {
+				rate_per_year,
+				compounding,
+			} => {
+				let new_rate = rate_per_year.ensure_add(rate)?;
+				Ok(InterestRate::Fixed {
+					rate_per_year: new_rate,
+					compounding,
+				})
+			}
+		}
+	}
+
+	pub fn ensure_sub(self, rate: Rate) -> Result<InterestRate<Rate>, ArithmeticError> {
+		match self {
+			InterestRate::Fixed {
+				rate_per_year,
+				compounding,
+			} => {
+				let new_rate = rate_per_year.ensure_sub(rate)?;
+				Ok(InterestRate::Fixed {
+					rate_per_year: new_rate,
+					compounding,
+				})
+			}
+		}
+	}
+}
+
 /// A trait that can be used to calculate interest accrual for debt
-pub trait InterestAccrual<InterestRate, Balance, Adjustment> {
+pub trait InterestAccrual<Rate, Balance, Adjustment> {
```

Comment:
Definitely, I would move now this trait and the `InterestRate` related types to their own module `cfg_traits::interest_rate`.